src/ICSharpCode.SharpZipLib/Zip/ZipInputStream.cs

```diff
@@ -503,10 +512,7 @@ private int ReadingNotSupported(byte[] destination, int offset, int count)
 		/// <returns>The actual number of bytes read.</returns>
 		private int InitialRead(byte[] destination, int offset, int count)
 		{
-			if (!CanDecompressEntry)
-			{
-				throw new ZipException("Library cannot extract this entry. Version required is (" + entry.Version + ")");
-			}
+			var usesDescriptor = (entry.Flags & (int)GeneralBitFlags.Descriptor) != 0;
```

Comment:
Can that use the HasFlag helper you added?

Reply:
There's a lot of places where it can be used, but this is essential the same as using that method. It might be a bit easier to read, but the main issue I had were the places where the flags where compared to a magic number instead. Really opaque to the reader.
Anyhow, I won't add any more changes of that type to this PR, but I'll instead merge it so that it can be used in subsequent PRs.